packages/react-server/src/ReactFlightServerConfigDebugNode.js

```diff
@@ -129,42 +147,50 @@ export function initAsyncDebugInfo(): void {
         }
         pendingOperations.set(asyncId, node);
       },
+      before(asyncId: number): void {
```

Comment:
I was trying to avoid implementing this phase which requires a lot more look ups into the map all the time.

However, it simplifies a lot by just putting the resolution here. It avoids all the special cases of when we have unresolved promises inside the thing that's currently being resolved and special cases for lazily filling in the IO end time.

What could happen before is that the end time for a already resolved await would be marked after it finished its `then` callback. However, that means that any awaits started inside of the `then` would get cut out since they started before the previously emitted await.